packages/commonwealth/client/scripts/views/pages/feed/index.tsx

```diff
@@ -5,108 +5,87 @@ import { ClassComponent } from 'mithrilInterop';
 import { CWText } from '../../components/component_kit/cw_text';
 import ErrorPage from '../error';
+import { Feed } from '../../components/feed';
 
 class FeedPage extends ClassComponent {
-  private isLoading: boolean;
-  private globalFeed: any;
-  private chainEvents: any;
-  private error: boolean;
-
   private getGlobalFeed = async () => {
     try {
       const activity = await fetchActivity(DashboardViews.Global);
-      this.globalFeed = activity.result
-        .map((item) => {
-          return {
-            ...item,
-            notificationData: item.notification_data,
-            createdAt: JSON.parse(item.notification_data).created_at,
-          };
-        })
-        .filter((item) => {
-          return (
-            JSON.parse(item.notification_data).chain_id === app.activeChainId()
-          );
-        });
+      const formattedData = activity.result
+        .map((item) => DashboardActivityNotification.fromJSON(item))
+        .filter(
+          (item) =>
+            JSON.parse(item.notificationData).chain_id === app.activeChainId()
+        );
+      return formattedData;
     } catch (err) {
-      this.error = true;
+      return err;
     }
   };
 
   private getChainEvents = async () => {
     try {
       const activity = await fetchActivity(DashboardViews.Chain);
-      this.chainEvents = activity.result
-        .map((item) => {
-          return {
-            blockNumber: item.block_number,
-            chain: item.chain,
-            eventData: item.event_data,
-            eventNetwork: item.event_network,
-            categoryId: 'chain-event',
-            createdAt: item.created_at,
-          };
-        })
-        .filter((item) => {
-          return item.chain === app.activeChainId();
-        });
+      const formattedData = activity.result
+        .map((item) => DashboardActivityNotification.fromJSON(item))
+        .filter((item) => item.chain === app.activeChainId());
+      return formattedData;
     } catch (err) {
-      this.error = true;
+      return err;
     }
   };
 
-  oninit() {
-    this.isLoading = true;
-    this.getGlobalFeed();
-    this.getChainEvents();
-    this.isLoading = false;
-  }
-
-  view() {
-    if (this.isLoading) {
-      return <PageLoading />;
-    }
+  private getCombinedFeed = async () => {
+    return Promise.all([this.getGlobalFeed(), this.getChainEvents()])
+      .then((result) => {
+        return {
+          result: this.sortFeed(result[0], result[1]),
+        };
+      })
+      .catch((err) => {
+        return err;
+      });
+  };
 
-    if (this.error) {
-      return <ErrorPage message="There was an error loading the feed." />;
+  private sortFeed = (globalFeed, chainEvents) => {
+    let sortedFeed = [];
+    if (globalFeed?.length > 0 && chainEvents?.length > 0) {
+      sortedFeed = globalFeed.concat(chainEvents).sort((a, b) => {
+        return (
+          new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
```

Comment:
In this case a built-on should be fine. No need to rewrite this, let's focus on shipping, small clean up like this can follow later (again, if we need it at all, in this case it looks like a built in is fine for the implementation.